resty

```diff
@@ -120,29 +120,39 @@ function resty-call() {
 	rm -f $tmpf
     fi
 
+    ## TODO: refactor block
     [ -n "$body" ] && [ "$body" != "@-" ] && [[ $# > 0 ]] && shift
     [ "$1" = "-Z" ] && raw="yes" && [[ $# > 0 ]] && shift
-    [ -n "$body" ] && opt="--data-binary"
-    [ "$method" = "HEAD" ] && opt="-I" && raw="yes"
-
-    ## Retrieve stored option in config dir.
-    ## §maybe: cache this when resty)
-    [ -f "$_RESTY_CONF_DIR/$domain" ] && eval "args2=( $(cat "$_RESTY_CONF_DIR/$domain" 2>/dev/null |sed 's/^ *//' |grep ^$method |cut -b $((${#method}+2))-) )"
-    local res=$((((curl -sLv $opt "$body" -X $method \
-		  -b "$cookies/$domain" -c "$cookies/$domain" \
-		  "${args2[@]}" "${curlopt2[@]}" "${curlopt[@]}" "$_path$query" \
-		    |sed 's/^/OUT /' && echo) 3>&2 2>&1 1>&3) \
-	       |sed 's/^/ERR /' && echo) 2>&1)
-    local out=$(echo "$res" |sed '/^OUT /s/^....//p; d')
-    local err=$(echo "$res" |sed '/^ERR /s/^....//p; d')
-    local ret=$(echo "$err" |sed \
-			   '/^.*HTTP\/1\.[01] [0-9][0-9][0-9]/s/.*\([0-9]\)[0-9][0-9].*/\1/p; d' \
-		 | tail -n1)
-    [ -n "$err" -a -n "$verbose" ] && echo "$err" 1>&2
+    [ -n "$body" ] && curl_opt="--data-binary"
+    [ "$method" = "HEAD" ] && curl_opt="-I" && raw="yes" # HERE??
+
+
+    if [ -f "$_RESTY_CONF_DIR/$domain" ] ; then  # Retrieve stored option in config dir.
+	## §maybe: cache this when resty)
+	local cmd="args2=( $(cat "$_RESTY_CONF_DIR/$domain" 2>/dev/null |sed 's/^ *//' |grep ^$method |cut -b $((${#method}+2))-) )"
+	# echo $cmd
+	eval $cmd
+    fi
+
+    # Launch command and retrieved streams
+    local res out err ret
+    res=$((((curl -sLv $curl_opt "$body" -X $method -b "$cookies/$domain" -c "$cookies/$domain" \
```

Comment:
indeed, but using rather the `-E` flag
simplified it a lot!

thanks for the suggestion